src/main/java/net/sf/jabref/logic/exporter/ModsExportFormat.java

```diff
@@ -1,58 +1,376 @@
+                            extent.setTotal(total);
+                            partDefinition.getDetailOrExtentOrDate().add(extent);
+                        }
+                    } else if (FieldName.URI.equals(key)) {
+                        addIdentifier(FieldName.URI, value, mods);
+                    } else if (FieldName.ISBN.equals(key)) {
+                        addIdentifier(FieldName.ISBN, value, mods);
+                    } else if (FieldName.ISSN.equals(key)) {
+                        addIdentifier(FieldName.ISSN, value, mods);
+                    } else if (FieldName.DOI.equals(key)) {
+                        addIdentifier(FieldName.DOI, value, mods);
+                    } else if (FieldName.PMID.equals(key)) {
+                        addIdentifier(FieldName.PMID, value, mods);
+                    } else if (FieldName.JOURNAL.equals(key)) {
+                        TitleInfoDefinition titleInfo = new TitleInfoDefinition();
+                        StringPlusLanguage title = new StringPlusLanguage();
+                        title.setValue(value);
+                        JAXBElement<StringPlusLanguage> element = new JAXBElement<>(new QName("title"),
+                                StringPlusLanguage.class, title);
+                        titleInfo.getTitleOrSubTitleOrPartNumber().add(element);
+                        relatedItem.getModsGroup().add(titleInfo);
+                    }
+
+                    addOriginInformation(key, value, originInfo);
+                }
+
+                relatedItem.getModsGroup().add(partDefinition);
+                relatedItem.setAtType("host");
+                mods.getModsGroup().add(relatedItem);
+                mods.getModsGroup().add(originInfo);
+                TypeOfResourceDefinition typeOfResource = new TypeOfResourceDefinition();
+                typeOfResource.setValue("text");
+                mods.getModsGroup().add(typeOfResource);
+                modsCollection.getMods().add(mods);
             }
-            finalizeSaveSession(ss, Paths.get(file));
-        } catch (IOException ex) {
+
+            JAXBContext context = JAXBContext.newInstance(ModsCollectionDefinition.class);
+            Marshaller marshaller = context.createMarshaller();
+            JAXBElement<ModsCollectionDefinition> jaxbElement = new JAXBElement<>(new QName("modsCollection"),
+                    ModsCollectionDefinition.class, modsCollection);
+
+            //since it has to be a prefix, use mods everywhere as prefix for elements
+            //see also http://www.loc.gov/standards/mods/v3/mods-userguide-intro.html
+            NamespacePrefixMapper myPrefixMapper = new NamespacePrefixMapper() {
+
+                @Override
+                public String getPreferredPrefix(String namespaceUri, String suggestion, boolean requirePrefix) {
+                    if ("http://www.loc.gov/mods/v3".equals(namespaceUri)) {
```

Comment:
Pleas consider extracting those as String constants, so in case of change they could be easily changed..